src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java

```diff
@@ -599,7 +603,7 @@ public CompoundNBT serializeNBT()
         compound.put(TAG_WORKER, workersTagList);
 
         compound.putInt(TAG_HIRING_MODE, this.hiringMode.ordinal());
-        @NotNull final ListNBT recipesTagList = recipes.stream()
+        @NotNull final ListNBT recipesTagList = recipes.stream().limit(this.getMaxRecipes() * 10)
             .map(iToken -> StandardFactoryController.getInstance().serialize(iToken))
```

Comment:
for anyone wondering, this is a limitter for the future, to avoid huge leaks, x10 should still be big enough to detect the leak for the player